pages/en/lb4/Server.md

```diff
@@ -0,0 +1,220 @@
+  export const EMITTER_EVENT = 'myserver.emitter.event';
+}
+```
+
+### Extend Context Class
+
+The Server class extends `Context` from `@loopback/context`.
+Its constructor takes in an `Application` instance as its parent `Context` and
+a configuration(optional) specific to a particular server instance.
+
+{% include code-caption.html content="myServer/src/MyServer.ts" %}
+```js
+import {Context} from '@loopback/context';
+import {Server} from '@loopback/core';
+
+export class MyServer extends Context implements Server {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) app: Application,
+    @inject(MyServerBindings.CONFIG) options?: MyServerConfig
+  ) {
+    super(app);
+  }  
+}
+
+export interface MyServerConfig {
+  x: string;
+  y: number;
+}
+```
+
+### Implement Server Functions
+
+You may have already noticed that the Server declaration above also implements a `Server` interface.
+There are two basic functions a Server must have:
+
+- start()
+- stop()
+
+`start` spins up the server to a working state, keep doing one or more things 
+and could interact with events from outside.
+
+`stop` stops a server by exiting its working state.
+
+{% include code-caption.html content="myServer/src/MyServer.ts" %}
+```js
+import {Server, Application, CoreBindings} from '@loopback/core';
+import {Context, inject} from '@loopback/context';
+import * as EventEmitter from 'events';
+
+class MyEmitter extends EventEmitter {}
+
+export class MyServer extends Context implements Server {
+  myEmitter: MyEmitter;
+
+  constructor(
+      @inject(CoreBindings.APPLICATION_INSTANCE) public app: Application
+  ) {
+    super(app);
+    // an Emitter listen on a certain event whose name is returned by `getEventName()`
+    this.myEmitter = new MyEmitter();
+  }
+
+  // return the event name
+  async getEventName(): Promise<string> {
+    return await this.get('MyServerBindings.EMITTER_EVENT');
+  }
+
+  async start(): Promise<void> {
+    console.log('Your server startted!');
+    let eventName = await this.getEventName();
+    // listen on an event and give some response
+    this.myEmitter.on(eventName, () => {
+      console.log('Your server is responding to an event...');
+    });
+    return Promise.resolve();
+  }
+
+  async stop(): Promise<void> {
+    let eventName = await this.getEventName();
+    // remove listeners
+    this.myEmitter.removeAllListeners(eventName);
+    console.log('Your server stopped!');
+    return Promise.resolve();
+  }
+}
+```
+
+### Bind a Server to Application
+
+Now you can bind a Server to your Application by calling `this.server()` 
+in the Application constructor.
+
+{% include code-caption.html content="application.ts" %}
+```js
+import { Application, ApplicationConfig } from '@loopback/core';
+import {MyServer} from './myServer/src/MyServer';
+
+export class MyApp extends Application {
+  constructor(options?: ApplicationConfig) {
+  super(options);
+  this.server(MyServer, 'serverFoo');
```

Comment:
Change the name of the instance. All instances bound in this way will have a key `servers.{name}` where `{name}` is the string provided. This would give you a key of `servers.serverFoo` which is unnecessary.